File: benchmarkDS/Vanilla/OPENAI_gpt-5-2025-08-07/Lse20.java
public class Lse20 {
    private static final int OFFSET = 1000;

    public static int addOffset(int value) {
        return value + OFFSET;
    }

    private static void runTests() {
        int[] tests = {0, 1, -1, 42, 999};
        for (int v : tests) {
            System.out.println("Test " + v + " -> " + addOffset(v));
        }
    }

    public static void main(String[] args) {
        if (args.length > 0) {
            try {
                int input = Integer.parseInt(args[0]);
                System.out.println(addOffset(input));
            } catch (NumberFormatException e) {
                System.out.println("Invalid integer input");
            }
        }
        runTests();
    }
}

File: benchmarkDS/CoT/CLAUDE_claude-sonnet-4-5-20250929/Lse20.cpp

#include <iostream>
#include <string>
#include <stdexcept>
#include <limits>

const int OFFSET = 1000;

int addOffset(const std::string& input) {
    try {
        size_t pos;
        long long value = std::stoll(input, &pos);
        
        // Check if entire string was parsed
        if (pos != input.length()) {
            throw std::invalid_argument("Invalid input: not a valid integer");
        }
        
        // Check for integer overflow
        if (value > std::numeric_limits<int>::max() - OFFSET || 
            value < std::numeric_limits<int>::min()) {
            throw std::overflow_error("Integer overflow detected");
        }
        
        return static_cast<int>(value) + OFFSET;
    } catch (const std::invalid_argument&) {
        throw std::invalid_argument("Invalid input: not a valid integer");
    } catch (const std::out_of_range&) {
        throw std::overflow_error("Integer overflow detected");
    }
}

int main() {
    // Test cases
    std::string testCases[] = {"100", "0", "-500", "2147482647", "50"};
    
    for (const auto& testCase : testCases) {
        try {
            int result = addOffset(testCase);
            std::cout << "Input: " << testCase << " -> Output: " << result << std::endl;
        } catch (const std::exception& e) {
            std::cout << "Input: " << testCase << " -> Error: " << e.what() << std::endl;
        }
    }
    
    return 0;
}
